numpy/lib/function_base.py

```diff
@@ -1323,21 +1323,15 @@ def piecewise(x, condlist, funclist, *args, **kw):
     """
     x = asanyarray(x)
     n2 = len(funclist)
-    if (isscalar(condlist) or not (isinstance(condlist[0], list) or
-                                   isinstance(condlist[0], ndarray))):
-        if not isscalar(condlist) and x.size == 1 and x.ndim == 0:
-            condlist = [[c] for c in condlist]
```

Comment:
Essentially, this line is replaced with `pass`, and then the nested `if`s are collapsed